service.go

```diff
@@ -111,14 +111,21 @@ func NewService(rootCa *x509.Certificate, ca *x509.Certificate, key crypto.Signe
 	}
 }
 
-// CaCerts return the CA certificate as per:
+// CaCerts return the root CA certificates as per:
 // https://www.rfc-editor.org/rfc/rfc7030.html#section-4.1.2
 func (s Service) CaCerts(ctx context.Context) ([]byte, error) {
-	bytes, err := pkcs7.DegenerateCertificate(s.rootCa.Raw)
-	if err != nil {
+	if envelope, err := pkcs7.NewSignedData(nil); err != nil {
 		return nil, err
+	} else {
+		for _, cert := range s.rootCa {
+			envelope.AddCertificate(cert)
+		}
+		if bytes, err := envelope.Finish(); err != nil {
+			return nil, err
+		} else {
+			return []byte(base64.StdEncoding.EncodeToString(bytes)), nil
+		}
```

Comment:
Fioconfig isn't currently accessing this stuff right? i.e it was just part of implementing EST?

Will we need to enforce or support anything around the `The EST server SHOULD include the three "Root CA Key Update"
   certificates OldWithOld, OldWithNew, and NewWithOld in the response
   chain.` wording in the spec?

Reply:
Further reading the spec got me to deduct that our proposed root CA update process is marginally different from what [EST spec says](https://datatracker.ietf.org/doc/html/rfc7030#section-4.1.3) and what the [CPM spec says](https://datatracker.ietf.org/doc/html/rfc4210#section-4.4.1):
- We return `OldWithOld` i.e. the old root CA.
- We return `NewWithNew` i.e. the new root CA.
- We return `NewWithOld` i.e. the new root CA cross-signed with old root CA.
- We don't return the `OldWithNew` i.e. the old root CA cross-signed with new root CA.

A rationale for `OldWithNew` from the spec is allowing to remove the old private key immediately after the phase 1.

I had a slightly different idea in mind.
As we work with IoT, which is unreliable by definition, we need a way to cancel or restart the root CA update procees.
So, we still need to keep the old private key around, until the root CA update is finished.
But that devoids the whole idea of `OldWithNew`, as the `OldWithOld` serves well enough.

Note that in our case for some of our customers the Factory Root CA is not a self-signed CA i.e. it is `OldWithHigherLevel`.
So, in the end the update process for them would look more like this: `OldWithHigherLevel`, `NewWithHigherLevel`, `NewWithOld`.
In this scenario the `OldWithNew` does not seem logical.

I suspect this is the very reason why the EST spec uses the word `SHOULD` unlike `MUST` on this topic.

Reply:
Having said that, I think we may return the `OldWithNew` certificate.

That has no added benefit for the `Fioconfig`, but we'd need to make a validation process in it a little bit more complex.
The EST spec says `The EST client MUST be able to handle these certificates in the response`.
So, probably, this is a good excercise to stay fully-compliant, anyway.

WDYT?

Reply:
Doesn't sound like it adds too much to the scope of work, so its probably worth it - if anything to avoid some weird support ticket someday.